### PR TITLE
PV Overhaul

### PR DESCRIPTION
**Pull request by kr45732**
Also tried to get rid of some duplicated code, unused methods, and deprecated methods

### CODE REVIEW

**Comment by kr45732** [comment]
PV works
uhm merge conflicts pain

**Comment by kr45732** [comment]
one more thing for the future is to move all the profile specific stuff that are cached in individual pages into the SkyblockProfiles class

**Review by ThatGravyBoat — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/CrimsonIslePage.java [hunk 1]
```diff
@@ -219,7 +223,7 @@ public void drawDojoStats(JsonObject data, int guiLeft, int guiTop) {
 
 		for (int i = 0; i < apiDojoTestNames.size(); i++) {
 			Utils.renderAlignedString(
-				apiDojoTestNames.get(apiDojoTestNames.keySet().toArray()[i]) + ": ",
+				new ArrayList<>(apiDojoTestNames.values()).get(i) + ": ",
```
[the remark this answers is not in the document]
Comment (kr45732) [reply src/main/java/io/github/moulberry/notenoughupdates/profileviewer/CrimsonIslePage.java]: Honestly not sure what's going on with the old code, refactored it.

**Comment by kr45732** [comment]
Ok tested PV, should be free of crashes/bugs now. Probably no further changes to this PR now.

**Comment by hannibal002** [comment]
`// This is to make infer happy but shouldn't be possible`
You can remove these comments.
I'm pretty sure `getSelectedProfile()` returns null immediately after a profile/world change.

**Comment by kr45732** [comment]
> `// This is to make infer happy but shouldn't be possible` You can remove these comments. I'm pretty sure `getSelectedProfile()` returns null immediately after a profile/world change.

This was since drawing all tasks are all called by the LevelPage which does the null check before drawing the tasks. But I can remove it.

File: src/main/java/io/github/moulberry/notenoughupdates/profileviewer/SkyblockProfiles.java [hunk 2]
```diff
@@ -0,0 +1,974 @@
...
+import java.util.Base64;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class SkyblockProfiles {
+	private static final String defaultNbtData = "Hz8IAAAAAAAAAD9iYD9kYD9kAAMAPwI/Gw0AAAA=";
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/profileviewer/SkyblockProfiles.java:54]: This still isnt valid GZIPped or non GZIPped NBT data, btw.

**Review by romangraef — APPROVED** [review]

**Pull request merged**